### PR TITLE
Feature: Improve codegen support for static assets and improve font loading for prototype-ui

### PR DESCRIPTION
**Pull request by Greenheart**
Changes in this PR:

* Improve font loading for `prototype-ui`.
* Use the Google Fonts css2 API to use latest best practices https://developers.google.com/fonts/docs/css2
* Use local hathora logo to reduce number of remote requests.

This was the original scope, but then I realized that plenty of non-handlebars files were loaded and passed through the compiler even they didn't use any handlebars features.

**Thus, additional changes in this PR include changes related to how the codegen handles static assets:**

* Only use handlebars for the template files that need it, and copy the rest directly.
* Slight codegen performance improvement
* Simplified template maintenance by clarifying which files actually use dynamic codegen, compared to being static assets/configs.
* Enable proper syntax highlighting for non-handlebars template files.

Feedback appreciated! :)

### CODE REVIEW

File: templates/base/client/prototype-ui/App.tsx.hbs [hunk 1]
```diff
@@ -84,7 +84,7 @@ function NavBar({ user }: { user?: UserData }) {
           <div className="flex-shrink-0 cursor-pointer" onClick={() => navigate(`/`)}>
             <img
               className="w-10 h-10"
-              src="https://i.ibb.co/sQMmTTq/hathora-hammer-logo-light.png"
+              src="/hathora-hammer-logo-light.png"
               alt="hathora-hammer-logo"
             />
```
Comment (hpx7) [inline templates/base/client/prototype-ui/App.tsx.hbs]: can you collapse this `img` tag into a single line now?

File: templates/base/client/prototype-ui/index.html.hbs [hunk 2]
```diff
@@ -7,9 +7,9 @@
...
     />
-    <link rel="stylesheet" href="//rsms.me/inter/inter.css" />
-    <link rel="stylesheet" href="//fonts.googleapis.com/css?family=Josefin+Sans:300,400,600,700,900&amp;lang=en" />
-    <link rel="stylesheet" href="//fonts.googleapis.com/css?family=Inconsolata:300,400,600,700,900&amp;lang=en" />
+    <link rel="preconnect" href="https://fonts.googleapis.com">
+    <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
+    <link href="https://fonts.googleapis.com/css2?family=Inconsolata:wght@300;400;600;700;900&family=Inter:wght@400;500;600;700&family=Josefin+Sans:wght@300;400;600;700&display=swap" rel="stylesheet">
```
Comment (hpx7) [inline templates/base/client/prototype-ui/index.html.hbs]:
```suggestion
    <link
      href="https://fonts.googleapis.com/css2?family=Inconsolata:wght@300;400;600;700;900&family=Inter:wght@400;500;600;700&family=Josefin+Sans:wght@300;400;600;700&display=swap"
      rel="stylesheet"
    />
```

**Comment by hpx7** [comment]
Just some minor formatting changes, otherwise this looks good!

**Comment by Greenheart** [comment]
Awesome, thanks! :smile:

File: templates/base/client/prototype-ui/index.html.hbs [hunk 3]
```diff
@@ -7,9 +7,12 @@
...
     />
-    <link rel="stylesheet" href="//rsms.me/inter/inter.css" />
-    <link rel="stylesheet" href="//fonts.googleapis.com/css?family=Josefin+Sans:300,400,600,700,900&amp;lang=en" />
-    <link rel="stylesheet" href="//fonts.googleapis.com/css?family=Inconsolata:300,400,600,700,900&amp;lang=en" />
+    <link rel="preconnect" href="https://fonts.googleapis.com">
```
Comment (hpx7) [inline templates/base/client/prototype-ui/index.html.hbs]:
```suggestion
    <link rel="preconnect" href="https://fonts.googleapis.com" />
```

File: templates/base/client/prototype-ui/index.html.hbs [hunk 4]
```diff
@@ -7,9 +7,12 @@
...
+    <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
```
Comment (hpx7) [inline templates/base/client/prototype-ui/index.html.hbs]:
```suggestion
    <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin />
```

File: templates/base/client/prototype-ui/index.html.hbs [hunk 5]
```diff
@@ -7,9 +7,12 @@
...
     />
-    <link rel="stylesheet" href="//rsms.me/inter/inter.css" />
-    <link rel="stylesheet" href="//fonts.googleapis.com/css?family=Josefin+Sans:300,400,600,700,900&amp;lang=en" />
-    <link rel="stylesheet" href="//fonts.googleapis.com/css?family=Inconsolata:300,400,600,700,900&amp;lang=en" />
+    <link rel="preconnect" href="https://fonts.googleapis.com" />
+    <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin />
+    <link
+      href="https://fonts.googleapis.com/css2?family=Inconsolata:wght@300;400;600;700;900&family=Inter:wght@400;500;600;700&family=Josefin+Sans:wght@300;400;600;700&display=swap"
+      rel="stylesheet"
+    />
```
Comment (hpx7) [inline templates/base/client/prototype-ui/index.html.hbs]:
```suggestion
    />
```

**Pull request merged**